conans/client/conf/__init__.py

```diff
@@ -119,3 +120,8 @@
 
 def get_default_settings_yml():
     return default_settings_yml
+
+
+def migrate_settings_file(cache, updater):
+    settings_path = cache.settings_path
+    updater(settings_path, get_default_settings_yml())
```

Comment:
Where is the ``updater()`` function?
I think the rename broke it, and it is not covered by tests?

In that case, it would be necessary to add some tests, use as inspirations the ones in conan 1.X, how the version.txt is forced to trigger an actual migration.